app/src/main/java/com/example/dine_and_donate/HomeFragments/NotificationsFragment.java

```diff
@@ -70,79 +50,46 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        allEvents = new ArrayList<>();
-        loaded = false;
     }
 
     @Override
     public void onViewCreated(@NonNull final View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
-        mRef = FirebaseDatabase.getInstance().getReference();
-        mRefForEvent = mRef.child("events");
-
         mRecyclerView = view.findViewById(R.id.notifications_rv);
-        mRecyclerView.setHasFixedSize(true);
-        LinearLayoutManager mLinearLayoutManager = new LinearLayoutManager(getContext());
-        mRecyclerView.setLayoutManager(mLinearLayoutManager);
         mNotificationList = new ArrayList<>();
         mNotificationsAdapter = new NotificationsAdapter(mNotificationList);
+        LinearLayoutManager mLinearLayoutManager = new LinearLayoutManager(getContext());
+        mRecyclerView.setLayoutManager(mLinearLayoutManager);
         mRecyclerView.setAdapter(mNotificationsAdapter);
+
         mNotificationList.clear();
+        loadTopPosts(0);
         Collections.reverse(mNotificationList);
         mNotificationsAdapter.notifyDataSetChanged();
```

Comment:
i didn't really review this code before, but something seems a little weird here. why do you set the notifs, then clear them, then reverse them? why not just do the Collections.reverse at the beginning, then add them to the adapter?